recipes/full_finetune_distributed.py

```diff
@@ -456,19 +458,55 @@ def _is_layer_fqn(s: str) -> bool:
         return model
 
     def _setup_optimizer(
-        self, cfg_optimizer: DictConfig, opt_state_dict: Optional[Dict[str, Any]] = None
+        self,
+        cfg_optimizer: DictConfig,
+        optimizer_in_bwd: bool = False,
+        opt_state_dict: Optional[Dict[str, Any]] = None,
     ) -> Optimizer:
-        optimizer = config.instantiate(cfg_optimizer, self._model.parameters())
-        if opt_state_dict:
-            training.load_from_full_optimizer_state_dict(
-                optimizer,
-                opt_state_dict,
-                self._device,
+        if not optimizer_in_bwd:
```

Comment:
This is a nit, but let's switch the order of this if/else so that it's `if optimizer_in_bwd`. Then it's closer to what's in the single-device recipe and easier to compare between the two

Reply:
We have the initial zero out at single device at:
https://github.com/pytorch/torchtune/blob/665ab3f0c5a68f024c4ce0c512138bf8aa062dff/recipes/full_finetune_single_device.py#L591-L593